README.rst

```diff
@@ -1874,823 +67,14 @@ as in the example below:
-optionals (we also have :code:`SELECT` statements within :code:`$match1` and
-:code:`$match2`), but it illustrates the the general idea.
-
-Performance Penalty
-^^^^^^^^^^^^^^^^^^^
-
-If we have many *compound* optionals in the given :code:`GraphQL`, the above
-procedure results in the union of a large number of :code:`MATCH` queries.
-Specifically, for :code:`n` compound optionals, we generate 2n different
-:code:`MATCH` queries. For each of the 2n subsets :code:`S` of the :code:`n`
-optional edges:
-
-- We remove the :code:`@optional` restriction for each traversal in :code:`S`.
-- For each traverse :code:`t` in the complement of :code:`S`, we entirely discard :code:`t` along
-  with all the vertices and directives within it, and we add a filter on the previous traverse to
-  ensure that the edge corresponding to :code:`t` does not exist.
-
-Therefore, we get a performance penalty that grows exponentially with
-the number of *compound* optional edges. This is important to keep in
-mind when writing queries with many optional directives.
-
-If some of those *compound* optionals contain :code:`@optional` vertex
-fields of their own, the performance penalty grows since we have to
-account for all possible subsets of :code:`@optional` statements that can be
-satisfied simultaneously.
-
-Optional :code:`type_equivalence_hints` parameter
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-This compilation parameter is a workaround for the limitations of the
-GraphQL and Gremlin type systems:
-
-- GraphQL does not allow :code:`type` to inherit from another :code:`type`, only to implement an
-  :code:`interface`.
-- Gremlin does not have first-class support for inheritance at all.
-
-Assume the following GraphQL schema:
-
-.. code::
-
-    type Animal {
-        name: String
-    }
-
-    type Cat {
-        name: String
-    }
-
-    type Dog {
-        name: String
-    }
-
-    union AnimalCatDog = Animal | Cat | Dog
-
-    type Foo {
-        adjacent_animal: AnimalCatDog
-    }
-
-An appropriate :code:`type_equivalence_hints` value here would be
-:code:`{ Animal: AnimalCatDog }`. This lets the compiler know that the
-:code:`AnimalCatDog` union type is implicitly equivalent to the :code:`Animal`
-type, as there are no other types that inherit from :code:`Animal` in the
-database schema. This allows the compiler to perform accurate type
-coercions in Gremlin, as well as optimize away type coercions across
-edges of union type if the coercion is coercing to the union's
-equivalent type.
-
-Setting :code:`type_equivalence_hints = { Animal: AnimalCatDog }` during
-compilation would enable the use of a :code:`@fold` on the
-:code:`adjacent_animal` vertex field of :code:`Foo`:
-
-.. code::
-
-    {
-        Foo {
-            adjacent_animal @fold {
-                ... on Animal {
-                    name @output(out_name: "name")
-                }
-            }
-        }
-    }
-
-SchemaGraph
-~~~~~~~~~~~
```

Comment:
I verified that this section is essentially copy paste except for some minor nits/improvements in ReadTheDocs.